Sources/SmithyHTTPClientAPI/HttpClientConfiguration.swift

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import AwsCommonRuntimeKit
+import struct Foundation.TimeInterval
+import enum Smithy.URIScheme
+import struct SmithyHTTPAPI.Headers
+
+public class HttpClientConfiguration {
```

Comment:
nit: `HTTPClientConfiguration` (note the caps)

As long as we're moving this to a new module, might as well make the naming consistent.